contracts/test/oracle/UniswapMathWrapper.t.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity >=0.4.0 <0.8.0;
+
+import {MockUniV3Pool} from "../../mock/MockUniV3Pool.sol";
+import {UniswapMathWrapper} from "../../oracle/uniswap/UniswapMathWrapper.sol";
+import {DSTest} from "../utils/DSTest.sol";
+
+contract UniswapMathWrapperTest is DSTest {
+  MockUniV3Pool private mockPool;
+  UniswapMathWrapper private uniswapMathWrapper;
+
+  uint32 private secondsAgo = 3;
+
+  function setUp() public {
+    // Mock has an observe() method that returns predefined test values. Used to ensure price > 0
+    mockPool = new MockUniV3Pool();
+    uniswapMathWrapper = new UniswapMathWrapper();
+  }
+
+  function testPrice() public {
+    uint256 price = uniswapMathWrapper.calculatePrice(address(mockPool), secondsAgo);
+    assertGt(price, 0);
```

Comment:
Difficult to unit test the exact price calculated. This function itself is just calling three Uniswap library methods for oracles:

```
(int24 arithmeticMeanTick, ) = OracleLibrary.consult(pool, secondsAgo);
uint160 sqrtPrice = TickMath.getSqrtRatioAtTick(arithmeticMeanTick);
uint256 price = FullMath.mulDiv(sqrtPrice, sqrtPrice, FixedPoint96.Q96);
```

Adding an e2e test to compare oracle performance with our primary oracle, Chainlink, as a means of validation. Also potentially worthwhile getting the implementation reviewed by a Uniswap team member